searchlib/src/main/java/com/yahoo/searchlib/rankingexpression/rule/SerializationContext.java

```diff
@@ -54,11 +55,11 @@ public SerializationContext(Collection<ExpressionFunction> functions, Map<String
         this(toMap(functions), bindings, serializedFunctions);
     }
 
-    private static ImmutableMap<String, ExpressionFunction> toMap(Collection<ExpressionFunction> list) {
-        ImmutableMap.Builder<String,ExpressionFunction> mapBuilder = new ImmutableMap.Builder<>();
+    private static Map<String, ExpressionFunction> toMap(Collection<ExpressionFunction> list) {
+        Map<String,ExpressionFunction> mapBuilder = new HashMap<>();
```

Comment:
The name 'mapBuilder' is confusing unless you know that it used to be an ImmutableMap.Builder. I suggest to use e.g. functionByName, name2Function, or simply functions.